src/controllers/trainingPlan/index.ts

```diff
@@ -0,0 +1,94 @@
+import { Request, Response } from 'express';
+import TrainingPlanService from '../../services/trainingPlan';
+import UserService from '../../services/user';
+import {
+  CreateTrainingPlanRequest,
+  FindTrainingPlanRequest,
+  PatchTrainingPlanRequest
+} from './types';
+import { handlePrismaError } from '../../lib/handle-prisma-error';
+import { decodeAccessToken } from '../../lib/jwt/decode-access-token';
+
+export default class TrainingPlanController {
+  static async getAll(_: Request, response: Response) {
+    const trainingPlans = await TrainingPlanService.getAll();
+    return response.success({ data: trainingPlans });
+  }
+
+  static async find(request: Request, response: Response) {
+    const { id } = request.params as any as FindTrainingPlanRequest;
+
+    const trainingPlan = await TrainingPlanService.find({ id });
+
+    if (!trainingPlan) {
+      return response.notfound();
+    }
+
+    return response.success({ data: trainingPlan });
+  }
+
+  static async create(request: Request, response: Response) {
+    const body = request.body as CreateTrainingPlanRequest;
+
+    try {
+      const loggedUser = await TrainingPlanController.authorizedUser(request);
```

Comment:
use routes middleware to check if the user is a trainer. remove this function `authorizedUser`

Reply:
if user is not allowed to access the route, then it should be blocked on the route and not on the controller.